### PR TITLE
tests/resource/aws_api_gateway_domain_name: Create and use EDGE endpoint CheckExists and CheckDestroy

### LINKED ISSUE #17420: tests/resource/aws_api_gateway_domain_name: Edge Endpoint Type CheckExists and CheckDestroy Missing
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Currently the `TestAccAWSAPIGatewayDomainName_CertificateArn` acceptance test fails in TeamCity:

```
=== CONT  TestAccAWSAPIGatewayDomainName_CertificateArn
resource_aws_api_gateway_domain_name_test.go:24: Step 1/2 error: Check failed: Check 1/6 error: NotFoundException: Invalid domain name identifier specified
--- FAIL: TestAccAWSAPIGatewayDomainName_CertificateArn (964.31s)
```

EDGE endpoint type resources can only exist in us-east-1 region. While the test function and configuration are set up to switch the region, the test is using `testAccCheckAWSAPIGatewayDomainNameDestroy` and `testAccCheckAWSAPIGatewayDomainNameExists` which use `testAccProvider` instead of `testAccProviderEc2Classic`. New CheckExists and CheckDestroy functions will need to be created using `testAccProviderApigatewayEdgeDomainName` and the test will need to be switched over to them.

### Affected Resource(s)

* aws_api_gateway_domain_name

### References

Similar:

* https://github.com/hashicorp/terraform-provider-aws/issues/17393

### PR DESCRIPTION
**Pull request by bflad**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #17420

Currently the `TestAccAWSAPIGatewayDomainName_CertificateArn` acceptance test fails in TeamCity:

```
=== CONT  TestAccAWSAPIGatewayDomainName_CertificateArn
resource_aws_api_gateway_domain_name_test.go:24: Step 1/2 error: Check failed: Check 1/6 error: NotFoundException: Invalid domain name identifier specified
--- FAIL: TestAccAWSAPIGatewayDomainName_CertificateArn (964.31s)
```

EDGE endpoint type resources can only exist in us-east-1 region. While the test function and configuration are set up to switch the region, the test is using `testAccCheckAWSAPIGatewayDomainNameDestroy` and `testAccCheckAWSAPIGatewayDomainNameExists` which use `testAccProvider` instead of `testAccProviderApigatewayEdgeDomainName`.

Output from acceptance testing in AWS Commercial:

```
--- PASS: TestAccAWSAPIGatewayDomainName_CertificateArn (972.57s)
--- PASS: TestAccAWSAPIGatewayDomainName_disappears (25.73s)
--- PASS: TestAccAWSAPIGatewayDomainName_MutualTlsAuthentication (198.61s)
--- PASS: TestAccAWSAPIGatewayDomainName_RegionalCertificateArn (249.05s)
--- PASS: TestAccAWSAPIGatewayDomainName_SecurityPolicy (129.72s)
--- PASS: TestAccAWSAPIGatewayDomainName_Tags (223.17s)
--- SKIP: TestAccAWSAPIGatewayDomainName_CertificateName (0.00s)
--- SKIP: TestAccAWSAPIGatewayDomainName_RegionalCertificateName (0.00s)
```

Output from acceptance testing in AWS GovCloud (US):

```
--- PASS: TestAccAWSAPIGatewayDomainName_disappears (42.47s)
--- PASS: TestAccAWSAPIGatewayDomainName_RegionalCertificateArn (80.03s)
--- PASS: TestAccAWSAPIGatewayDomainName_SecurityPolicy (171.64s)
--- PASS: TestAccAWSAPIGatewayDomainName_Tags (112.22s)
--- SKIP: TestAccAWSAPIGatewayDomainName_CertificateArn (0.00s)
--- SKIP: TestAccAWSAPIGatewayDomainName_CertificateName (0.00s)
--- SKIP: TestAccAWSAPIGatewayDomainName_MutualTlsAuthentication (0.00s)
--- SKIP: TestAccAWSAPIGatewayDomainName_RegionalCertificateName (0.00s)
```

### CODE REVIEW

File: aws/resource_aws_api_gateway_domain_name_test.go [hunk 1]
```diff
@@ -24,12 +25,12 @@ func TestAccAWSAPIGatewayDomainName_CertificateArn(t *testing.T) {
 	resource.ParallelTest(t, resource.TestCase{
 		PreCheck:          func() { testAccPreCheck(t); testAccPreCheckApigatewayEdgeDomainName(t) },
 		ProviderFactories: testAccProviderFactories,
-		CheckDestroy:      testAccCheckAWSAPIGatewayDomainNameDestroy,
+		CheckDestroy:      testAccCheckAWSAPIGatewayEdgeDomainNameDestroy,
```
Comment (anGie44) [inline aws/resource_aws_api_gateway_domain_name_test.go:28]: aside: maybe updating the test name and config could be helpful too when going back to understand why these use `Edge` related funcs at first glance --  `TestAccAWSAPIGatewayDomainName_EdgeCertificateArn` and `testAccAWSAPIGatewayDomainNameConfig_EdgeCertificateArn`

**Review by anGie44 — APPROVED** [review]
Looks great 👍  -- just a small comment related to pre-existing naming convention

File: aws/resource_aws_api_gateway_domain_name_test.go (hunk 1 above)
Comment (bflad) [reply aws/resource_aws_api_gateway_domain_name_test.go:28]: Happy to use whatever convention you all feel is appropriate. The current test naming is based on the root attribute being tested so `certificate_arn` (which requires `endpoint_configuration` `types` `EDGE` but theoretically could be named `_EndpointConfiguration_Types_Edge`) versus `regional_certificate_arn` (which requires `endpoint_configuration` `types` `REGIONAL` but theoretically could be named `_EndpointConfiguration_Types_Regional`) etc.

Comment (bflad) [reply aws/resource_aws_api_gateway_domain_name_test.go:28]: If you have continuing feels on this, please create a new issue and let's discuss!

**Pull request merged**